core/src/test/scala/org/apache/spark/scheduler/MapStatusSuite.scala

```diff
@@ -128,4 +130,23 @@ class MapStatusSuite extends SparkFunSuite {
     assert(size1 === size2)
     assert(!success)
   }
+
+  test("Blocks which are bigger than 2 * average size should not be underestimated.") {
+    val sizes = Array.concat(Array.fill[Long](1000)(1L), (1000L to 2000L).toArray)
```

Comment:
can we test just several blocks? e.g. `val sizes = Array(1L, 2L, 1000L, 2000L)`, then we can check all the expected sizes.